### DISCUSSION TITLE
Changing helm values in kube.tf is not recognized

### DISCUSSION THREAD

**Discussion by hesyar** [mysticaltech/terraform-hcloud-kube-hetzner#431] [Q&A] [answered]
When I change the helm values of cert-manager, lets say replicas from 3 to 4, and then do an apply, terraform claims that it couldn't find a change and the config seems to be same as on the servers but it isn't .

Expectation is that it knows rhat I changed that value and it should apply.

**Comment by mysticaltech** [comment, marked as answer]
Indeed @hesyar, the helm values via terraform apply just on first run (we will edit the doc to clarify that). To do it, the best way to change them is via an `helmchartconfig` (more on this in the readme).

In your case, pull down the helmchart with `kubectl get helmchart -A`, find the name, and `kubectl get helmchart <name> -n <namespace> -o yaml > cert-manager-helmchartconfig.yaml`, then edit the resulting file by changing it to a helmchartconfig, both definitions follow the same format, just the custom resource name changes, see https://docs.k3s.io/helm. 

If your case, you want to edit the helm values, so it goes under valuescontent, under the right section that you find in the helm values file.

Moving this to a discussion. Please let me know how it goes. 

_P.S. Always good to know about the `kubectl get crds` command, as you can really dive deep down how the whole cluster works._

**Comment by hesyar** [comment]
Hi @mysticaltech thanks a lot. Got, it, I thought it would be better to have it configured through terraform but yes, applying the HelmChartConfig works properly. Thanks again!

**Reply by mysticaltech** [reply]
Thanks, @hesyar! If you can mark the above as answer, it would be great! 🙏
